presentation-compiler/src/main/dotty/tools/pc/SymbolInformationProvider.scala

```diff
@@ -0,0 +1,128 @@
+package scala.meta.internal.pc
+
+import scala.util.control.NonFatal
+
+import scala.meta.pc.PcSymbolKind
+import scala.meta.pc.PcSymbolProperty
+
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Denotations.Denotation
+import dotty.tools.dotc.core.Denotations.MultiDenotation
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.core.Names.*
+import dotty.tools.dotc.core.StdNames.nme
+import dotty.tools.dotc.core.Symbols.*
+import dotty.tools.pc.utils.MtagsEnrichments.metalsDealias
+import dotty.tools.pc.SemanticdbSymbols
+
+class SymbolInformationProvider(using Context):
+  private def toSymbols(
+      pkg: String,
+      parts: List[(String, Boolean)],
+  ): List[Symbol] =
+    def collectSymbols(denotation: Denotation): List[Symbol] =
+      denotation match
+        case MultiDenotation(denot1, denot2) =>
+          collectSymbols(denot1) ++ collectSymbols(denot2)
+        case denot => List(denot.symbol)
+
+    def loop(
+        owners: List[Symbol],
+        parts: List[(String, Boolean)],
+    ): List[Symbol] =
+      parts match
+        case (head, isClass) :: tl =>
+          val foundSymbols =
+            owners.flatMap { owner =>
+              val next =
+                if isClass then owner.info.member(typeName(head))
+                else owner.info.member(termName(head))
+              collectSymbols(next).filter(_.exists)
+            }
+          if foundSymbols.nonEmpty then loop(foundSymbols, tl)
+          else Nil
+        case Nil => owners
+
+    val pkgSym =
+      if pkg == "_empty_" then requiredPackage(nme.EMPTY_PACKAGE)
+      else requiredPackage(pkg)
+    loop(List(pkgSym), parts)
+  end toSymbols
+
+  def info(symbol: String): Option[PcSymbolInformation] =
+    val index = symbol.lastIndexOf("/")
+    val pkg = normalizePackage(symbol.take(index + 1))
+
+    def loop(
+        symbol: String,
+        acc: List[(String, Boolean)],
+    ): List[(String, Boolean)] =
+      if symbol.isEmpty() then acc.reverse
+      else
+        val newSymbol = symbol.takeWhile(c => c != '.' && c != '#')
+        val rest = symbol.drop(newSymbol.size)
+        loop(rest.drop(1), (newSymbol, rest.headOption.exists(_ == '#')) :: acc)
+    val names =
+      loop(symbol.drop(index + 1).takeWhile(_ != '('), List.empty)
+
+    val foundSymbols =
+      try toSymbols(pkg, names)
+      catch case NonFatal(e) => Nil
+
+    val (searchedSymbol, alternativeSymbols) =
+      foundSymbols.partition(compilerSymbol =>
```

Comment:
This can look better with fewer braces :D :
```scala
foundSymbols.partition: compilerSymbol =>
  SemanticdbSymbols.symbolName(compilerSymbol) == symbol